src/server/enrollment-system.js

```diff
@@ -37,19 +37,169 @@ const formTemplate = {
   }
 };
 
+/**
+ * Returns a string left-padded with '0' of the given width.
+ *
+ * Warning: If 'number' is wider than 'width' characters, only the right-most 'width'
+ * characters are returned.
+ *
+ * TODO: should this explicitly check if 'number' and 'width' are actually numbers?
+ *
+ * @example
+ * // returns "0001"
+ * zeroPadNumber(1, 4);
+ * @example
+ * // returns "15"
+ * zeroPadNumber(31415, 2);
+ *
+ * @param {number} number The number you want left-padded with '0' characters.
+ * @param {number} width The width of the resulting string.
+ * @returns {String}
+ */
 // TODO(awong): Move to validations and add unittests.
 function zeroPadNumber(number, padding) {
   return (new Array(padding + 1).join('0') + number).slice(-padding);
 }
 
-// TODO(awong): Move to validations and add unittests.
-function formDateToESDate(dateObject) {
-  if (dateObject.month >= 1 && dateObject.month <= 12 &&
-    dateObject.day >= 1 && dateObject.month <= 31 && // TODO: how robust does this need to be? Form validates as well.
-    dateObject.year > 1) {
-    return `${zeroPadNumber(dateObject.month, 2)}/${zeroPadNumber(dateObject.day, 2)}/${zeroPadNumber(dateObject.year, 4)}`;
+/**
+ * Returns true when the provided year is a leap-year in the Gregorian calendar.
+ *
+ * Based on "Introduction To Calendars" paragraph at http://aa.usno.navy.mil/faq/docs/calendars.php
+ * specifically, the paragraph:
+ *
+ * "The Gregorian Calendar has become the internationally accepted civil calendar. The
+ * leap year rule for the Gregorian Calendar differs slightly from one for the Julian
+ * Calendar. The Gregorian leap year rule is: Every year that is exactly divisible by four is
+ * a leap year, except for years that are exactly divisible by 100, but these centurial
+ * years are leap years if they are exactly divisible by 400. For example, the years 1700, 1800,
+ * and 1900 are not leap years, but the year 2000 is. ..."
+ *
+ * @example
+ * // returns:
+ * [ [ 1700, false ],
+ *   [ 1800, false ],
+ *   [ 1900, false ],
+ *   [ 2000, true ],
+ *   [ 2011, false ],
+ *   [ 2012, true ],
+ *   [ 2013, false ],
+ *   [ 2014, false ],
+ *   [ 2015, false ],
+ *   [ 2016, true ],
+ *   [ 2017, false ] ]
+ *
+ * [1700,1800,1900,2000,2011,2012,2013,2014,2015,2016,2017].map((a)=>{return [a,isLeapYear(a)]})
+ *
+ * TODO: Albert, please double check the date logic.
+ *
+ * @param {number} year The year you want to test.
+ * @returns {boolean}
+ */
+function isLeapYear(year) {
```

Comment:
https://github.com/moment/momentjs.com/blob/master/docs/moment/05-query/09-is-leap-year.md